crac-plugin/src/main/java/io/micronaut/gradle/crac/MicronautCRaCPlugin.java

```diff
@@ -0,0 +1,247 @@
+package io.micronaut.gradle.crac;
+
+import com.bmuschko.gradle.docker.tasks.container.*;
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage;
+import com.bmuschko.gradle.docker.tasks.image.DockerPushImage;
+import com.bmuschko.gradle.docker.tasks.image.Dockerfile;
+import io.micronaut.gradle.MicronautExtension;
+import io.micronaut.gradle.MicronautRuntime;
+import io.micronaut.gradle.PluginsHelper;
+import io.micronaut.gradle.crac.tasks.CheckpointScriptTask;
+import io.micronaut.gradle.docker.DockerBuildStrategy;
+import io.micronaut.gradle.docker.model.MicronautDockerImage;
+import io.micronaut.gradle.docker.tasks.BuildLayersTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.plugins.ExtensionContainer;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.TaskProvider;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static io.micronaut.gradle.Strings.capitalize;
+
+public class MicronautCRaCPlugin implements Plugin<Project> {
+
+    public static final String CRAC_DEFAULT_BASE_IMAGE = "ubuntu:18.04";
+    public static final String CRAC_DEFAULT_BASE_IMAGE_PLATFORM = "linux/amd64";
+    private static final String CRAC_TASK_GROUP = "CRaC";
+    public static final String CRAC_CHECKPOINT = "crac_checkpoint";
+    public static final String BUILD_DOCKER_DIRECTORY = "docker/";
+
+    @Override
+    public void apply(Project project) {
+        configurePlugin(project);
+    }
+
+    private void configurePlugin(Project project) {
+        ExtensionContainer extensions = project.getExtensions();
+        MicronautExtension micronautExtension = extensions.getByType(MicronautExtension.class);
+        CRaCConfiguration configuration = createCRaCConfiguration(project);
+        NamedDomainObjectContainer<MicronautDockerImage> dockerImages = (NamedDomainObjectContainer<MicronautDockerImage>) micronautExtension.getExtensions().findByName("dockerImages");
+        createCheckpointDockerImage(project, dockerImages.findByName("main"), configuration);
+    }
+
+    private CRaCConfiguration createCRaCConfiguration(Project project) {
+        MicronautExtension micronautExtension = PluginsHelper.findMicronautExtension(project);
+        CRaCConfiguration crac = micronautExtension.getExtensions().create("crac", CRaCConfiguration.class);
+        crac.getEnabled().convention(true);
+        crac.getImageName().convention("Woo");
```

Comment:
Is this a real name?

Reply:
OOpps...

It was vestigial...  Removed 👍